cmd/arktos-network-controller/app/dns_deployment.go

```diff
@@ -265,6 +265,10 @@ func ensureToCreateDeployment(net *v1.Network, client kubernetes.Interface) erro
 									"-conf",
 									"/etc/coredns/Corefile",
 								},
+								Env: []corev1.EnvVar{
+									{Name: "KUBERNETES_SERVICE_HOST", Value: kubeAPIServerIP},
```

Comment:
It is strange that api server ip address is saved in deployment spec. Ideally, API server would be accessed via load balancer and dynamically allocated during system run. It could be changing as system maintenance is going on. Put it into spec make the data easily to stale and problematic even in short period.

Reply:
this workaround is temporary fix to unblock mizar integration; the long-term solution should, as you had guessed out, use kubernetes service ip instead - issue #745 opened to track.

For now, we cannot adopt k8s service ip directly, as it would break flat-type networks.